### PR TITLE
fix: return always actual ui langiage for users

### PR DESCRIPTION
**Pull request by tikhanovichA**
https://oat-sa.atlassian.net/browse/AUT-3512

In case we use key value storage for users we need to return the actual value from the storage

How to test:
1. Go to `My settings`
2. Update the language
3. Check that all UI language was updated after changing 

Can be tested on http://test-authoring-aut-3512.playground.kitchen.it.taocloud.org:40981/

### CODE REVIEW

File: src/AuthKeyValueUser.php [hunk 1]
```diff
@@ -159,27 +157,14 @@ public function getUserRawParameters()
         return $this->userRawParameters;
     }
 
-    /**
-     * @param mixed $language
-     */
-    public function setLanguageUi($languageUri)
```
Comment (andreluizmachado) [inline src/AuthKeyValueUser.php:165]: Are we sure we are not using this method anywhere else?

File: src/AuthKeyValueUser.php [hunk 2]
```diff
@@ -159,27 +157,14 @@ public function getUserRawParameters()
...
-        }
-
-        return $this;
-    }
-
     /**
      * @return array
      */
-    public function getLanguageUi()
+    public function getLanguageUiFromParams($params)
```
Comment (andreluizmachado) [inline src/AuthKeyValueUser.php:163]: can we specify the return type hint?

File: src/AuthKeyValueUser.php [hunk 3]
```diff
@@ -277,4 +258,14 @@ protected function getAuthKeyValueUserService()
     {
         return ServiceManager::getServiceManager()->get(AuthKeyValueUserService::SERVICE_ID);
     }
+
+    /**
+     * @return array
+     */
+    private function getKeyValueUserData() {
+        $login = current($this->getPropertyValues(GenerisRdf::PROPERTY_USER_LOGIN));
```
Comment (andreluizmachado) [inline src/AuthKeyValueUser.php:266]: current might return false. is it ok here?

File: src/AuthKeyValueUser.php [hunk 4]
```diff
@@ -277,4 +258,14 @@ protected function getAuthKeyValueUserService()
...
+        $userData = $this->getAuthKeyValueUserService()->getUserData($login);
+        $params = json_decode($userData[AuthKeyValueUserService::USER_PARAMETERS],true);
```
Comment (andreluizmachado) [inline src/AuthKeyValueUser.php:268]: is it possible that `AuthKeyValueUserService::USER_PARAMETERS`  might not exist in the array?

File: src/AuthKeyValueUser.php [hunk 5]
```diff
@@ -216,7 +201,8 @@ public function getPropertyValues($property)
                     $returnValue = $this->getLanguageDefLg();
                     break;
                 case GenerisRdf::PROPERTY_USER_UILG :
```
Comment (andreluizmachado) [inline src/AuthKeyValueUser.php:203]: can we use a `match` statement here?

File: src/AuthKeyValueUser.php (hunk 1 above)
Comment (tikhanovichA) [reply src/AuthKeyValueUser.php:165]: yep, it was used only in https://github.com/oat-sa/generis-auth-keyvalue/pull/31/files#diff-1627d8bb1176ffa5aee77d1d5c585c37ee07dd181b3a57447b248d5a457a1b33L95

File: src/AuthKeyValueUser.php (hunk 3 above)
Comment (tikhanovichA) [reply src/AuthKeyValueUser.php:214]: fixed

File: src/AuthKeyValueUser.php (hunk 2 above)
Comment (tikhanovichA) [reply src/AuthKeyValueUser.php]: fixed

File: src/AuthKeyValueUser.php (hunk 4 above)
Comment (tikhanovichA) [reply src/AuthKeyValueUser.php]: fixed

File: src/AuthKeyValueUser.php (hunk 5 above)
Comment (tikhanovichA) [reply src/AuthKeyValueUser.php]: fixed

**Pull request merged**